diff -up android-tools-34.0.4/vendor/adb/client/incremental_utils.cpp.include android-tools-34.0.4/vendor/adb/client/incremental_utils.cpp
--- android-tools-34.0.4/vendor/adb/client/incremental_utils.cpp.include	2023-09-03 07:06:16.000000000 -0600
+++ android-tools-34.0.4/vendor/adb/client/incremental_utils.cpp	2024-02-27 19:25:25.026823079 -0700
@@ -24,6 +24,7 @@
 #include <ziparchive/zip_archive.h>
 #include <ziparchive/zip_writer.h>
 
+#include <algorithm>
 #include <array>
 #include <cinttypes>
 #include <numeric>
diff -up android-tools-34.0.4/vendor/core/fs_mgr/liblp/super_layout_builder.cpp.include android-tools-34.0.4/vendor/core/fs_mgr/liblp/super_layout_builder.cpp
--- android-tools-34.0.4/vendor/core/fs_mgr/liblp/super_layout_builder.cpp.include	2023-09-03 07:06:37.000000000 -0600
+++ android-tools-34.0.4/vendor/core/fs_mgr/liblp/super_layout_builder.cpp	2024-02-27 19:31:26.527846061 -0700
@@ -17,6 +17,8 @@
 
 #include <liblp/liblp.h>
 
+#include <algorithm>
+
 #include "images.h"
 #include "utility.h"
 #include "writer.h"
diff -up android-tools-34.0.4/vendor/core/fs_mgr/liblp/utility.cpp.include android-tools-34.0.4/vendor/core/fs_mgr/liblp/utility.cpp
--- android-tools-34.0.4/vendor/core/fs_mgr/liblp/utility.cpp.include	2023-09-03 07:06:37.000000000 -0600
+++ android-tools-34.0.4/vendor/core/fs_mgr/liblp/utility.cpp	2024-02-27 19:42:57.193703638 -0700
@@ -25,6 +25,7 @@
 #include <sys/ioctl.h>
 #endif
 
+#include <algorithm>
 #include <map>
 #include <string>
 #include <vector>
